### PR TITLE
#3 Use django app to query cve-search and display cve-ids

### PR DESCRIPTION
**Pull request by kartiksibal**
Signed-off-by: Kartik Sibal <kartiksibal@gmail.com>

### CODE REVIEW

File: app/vulncode_app/views.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+def product(request, product):
+    """
+    Queries the cve-search api with just
+    a product name.
+    """
+    data = api.output_cve_id(name=product)
+    return HttpResponse(data)
+
+def product_ver(request, product, ver):
```
Comment (tdruez) [inline app/vulncode_app/views.py]: Why using `version` in the api module and `ver` here?
Always prefer to use easy to read and unambiguous variable name.
Also be consistent with your naming within the codebase,  if the content is the same, use the same name.
`def product_ver(request, product, ver):`
->
`def product_version(request, product, version):`

File: app/vulncode_app/views.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+from django.http import HttpResponse
+from django.shortcuts import render
+
+import vulncode_app.api_data as api
+
+def product(request, product):
```
Comment (tdruez) [inline app/vulncode_app/views.py]: You are using `package` in the api_data module and `product` here, any particular reason?

File: app/vulncode_app/views.py (hunk 1 above)
Comment (kartiksibal) [inline app/vulncode_app/views.py]: I had primarily done it to sort of create a little differentiation, b/w what's being passed and what it is being checked to. But, your point is more valid. 😄

File: app/vulncode_app/views.py (hunk 2 above)
Comment (kartiksibal) [inline app/vulncode_app/views.py]: Somewhat in contrast to cve-search. I'd switch 'em to package.

File: app/vulncode_app/views.py (hunk 1 above)
Comment (tdruez) [inline app/vulncode_app/views.py]: `sort of create a little differentiation` should be avoided, consistency first! :)

Comment (kartiksibal) [inline app/vulncode_app/views.py]: @tdruez Gotcha! 😉

File: app/vulncode_app/views.py [hunk 3]
```diff
@@ -1,26 +1,27 @@
...
     Queries the cve-search api with a package
     name and version.
     """
-    api.output_cve_id(name=name, version=version)
+    raw_data = api.data_cve_circl(name=name, version=version)
+    fields_names = ['id', 'summary', 'cvss']
+    extracted_data = api.extract_fields(raw_data, fields_names, version=True)
 
-    return JsonResponse({'cve_id':api.ids,'summary':api.summary,'cvss':api.cvss})
+    return HttpResponse(json.dumps(extracted_data))
```
Comment (kartiksibal) [inline app/vulncode_app/views.py]: @pombredanne I had changed the data structure of the data, from lists to dictionary and this is sort of what I had implemented.

File: .cache/v/cache/lastfailed [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (pombredanne) [inline .cache/v/cache/lastfailed]: Do not commit the .cache dir: it should be removed and the git ignored.

File: app/app/settings.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
+"""
```
Comment (pombredanne) [inline app/app/settings.py]: Missing license header

**Pull request merged**